### PR TITLE
Dropped Kubernetes 1.16 from relevant docs

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/419

### CODE REVIEW

File: enterprise/v0.16/09_resources/04_version-compatibility-reference.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ It's worth noting that while the tables below reference the minimum compatible v
 | ------------------- | ---------------- | ---- | ------------ | -------- | ----------------------------------------------------------------------- | --------------| -------------- |
 | v0.16               | 1.16, 1.17, 1.18 | 3    | 0.12, 0.13.5 | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14                               | 3.6, 3.7, 3.8 | 0.16           |
 | v0.23               | 1.16, 1.17, 1.18 | 3    | 0.13.5       | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 1.10.15, 2.0.0, 2.0.2, 2.1.0 | 3.6, 3.7, 3.8 | 0.23           |
-| v0.25               | 1.16, 1.17, 1.18, 1.19, 1.20 | 3    | 0.13.5       | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 1.10.15, 2.0.0, 2.0.2, 2.1.0 | 3.6, 3.7, 3.8 | 0.25           |
+| v0.25               | 1.17, 1.18, 1.19, 1.20 | 3    | 0.13.5       | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 1.10.15, 2.0.0, 2.0.2, 2.1.0 | 3.6, 3.7, 3.8 | 0.25           |
```
Comment (danielhoherd) [inline enterprise/v0.16/09_resources/04_version-compatibility-reference.md:19]: FWIW this line messes up the table formatting. Do we have a `prettier` hook or something that will fix this?

File: enterprise/v0.23/10_resources/04_version-compatibility-reference.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ It's worth noting that while the tables below reference the minimum compatible v
 | ------------------- | ---------------- | ---- | ------------ | -------- | ----------------------------------------------------------------------- | --------------| -------------- |
 | v0.16               | 1.16, 1.17, 1.18 | 3    | 0.12, 0.13.5 | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14                               | 3.6, 3.7, 3.8 | 0.16           |
 | v0.23               | 1.16, 1.17, 1.18 | 3    | 0.13.5       | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 1.10.15, 2.0.0, 2.0.2, 2.1.0 | 3.6, 3.7, 3.8 | 0.23           |
-| v0.25               | 1.16, 1.17, 1.18, 1.19, 1.20 | 3    | 0.13.5       | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 1.10.15, 2.0.0, 2.0.2, 2.1.0 | 3.6, 3.7, 3.8 | 0.25           |
+| v0.25               | 1.17, 1.18, 1.19, 1.20 | 3    | 0.13.5       | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 1.10.15, 2.0.0, 2.0.2, 2.1.0 | 3.6, 3.7, 3.8 | 0.25           |
```
Comment (danielhoherd) [inline enterprise/v0.23/10_resources/04_version-compatibility-reference.md:19]: Same formatting issue here.

File: enterprise/v0.23/06_manage-astronomer/07_upgrade-to-0-25.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ A few important notes before you start:
 
 Ensure that the following software is updated to the appropriate version:
 
-- **Kubernetes**: Your version must be 1.16, 1.17, or 1.18. If you need to upgrade Kubernetes, contact your cloud provider's support or your Kubernetes administrator.
+- **Kubernetes**: Your version must be 1.17 or 1.18. If you need to upgrade Kubernetes, contact your cloud provider's support or your Kubernetes administrator.
```
Comment (danielhoherd) [inline enterprise/v0.23/06_manage-astronomer/07_upgrade-to-0-25.md:22]: I'm not sure we deprecated 1.16 on release 0.23. Is that what we plan to do? AFAIK we still support it. We haven't removed 1.16 anything from the release-0.23 branch, only the release-0.25 branch. https://github.com/astronomer/astronomer/blob/release-0.23/.circleci/generate_circleci_config.py#L13

File: enterprise/v0.23/10_resources/04_version-compatibility-reference.md (hunk 2 above)
Comment (danielhoherd) [reply enterprise/v0.23/10_resources/04_version-compatibility-reference.md:19]: We do not support 1.19 or 1.20 on release-0.23, only up to release-0.18

File: enterprise/v0.16/09_resources/04_version-compatibility-reference.md (hunk 1 above)
Comment (jwitz) [reply enterprise/v0.16/09_resources/04_version-compatibility-reference.md:19]: Yep I can run it after the content is finalized, it'll restructure the table with perfect spacing

**Comment by danielhoherd** [comment]
I don't know what's going on but Github is showing me duplicate conversations in this issue. Sorry if I somehow caused that.

File: enterprise/v0.23/06_manage-astronomer/07_upgrade-to-0-25.md (hunk 3 above)
Comment (jwitz) [reply enterprise/v0.23/06_manage-astronomer/07_upgrade-to-0-25.md:22]: I assumed that we'd have a prerequisite of not using 1.16 for upgrading to v0.25 from v0.23. When customers upgrade, they always upgrade to the latest patch version of the major release right?

Comment (danielhoherd) [reply enterprise/v0.23/06_manage-astronomer/07_upgrade-to-0-25.md:22]: ok, yes, we do need astronomer upgrades to occur on a version of k8s that is supported by the target astronomer version. good call.

**Review by danielhoherd — APPROVED** [review]

**Pull request merged**